collector/cpu_linux.go

```diff
@@ -79,8 +80,11 @@ func NewCPUCollector(logger log.Logger) (Collector, error) {
 
 	isolcpus, err := sysfs.IsolatedCPUs()
 	if err != nil {
-		level.Warn(logger).Log("msg", "Unable to get isolated cpus, defaulting to []")
-		isolcpus = []uint16{}
+		if os.IsNotExist(err) {
```

Comment:
One tiny last nit: In golang early returns are preferred in such situation. Can you do this instead?
```
if !os.IsNotExist(err) {
  return nil, fmt.Errorf("Unable to get isolated cpus: %w", err)
}
level.Debug(logger).Log("msg", "Could not open isolated file", "error", err)
```